docs/dev/action_collection.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Action Collection
+---
+
```

Comment:
a small bulleted list of what the action collection hopes to achieve would be super useful. You get it once you read the whole thing and think it through + have the historical context of problems that zap had, but it would be put into just a few bullets that would make this whole thing much more useful